### PR TITLE
Technical/added api for match not matched accommodations

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs [hunk 1]
```diff
@@ -231,6 +231,24 @@ public async Task<IActionResult> MatchUncertain(int uncertainMatchId)
...
+        /// <summary>
+        /// Matches two ht accommodations
+        /// </summary>
+        /// <param name="htId"></param>
+        /// <param name="htIdToMatch"></param>
+        /// <returns></returns>
+        [HttpPost("accommodations/match")]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> MatchAccommodations(int htId, int htIdToMatch)
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:243]: Maybe think about more descriptive names?
As I understood, first is main and the second is something that is merged to main.
Maybe sourceHtId or originalHtId for the first arg?

Comment (mingazhev) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:243]: If there is no logical difference, maybe just htId1, htId2?

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs [hunk 2]
```diff
@@ -116,7 +118,80 @@ public async Task<Result> RecalculateData(int id)
...
+        {
+            var accommodation =
+                await _context.Accommodations.SingleOrDefaultAsync(ac => ac.Id == htId && ac.IsActive);
+            var accommodationToMatch =
+                await _context.Accommodations.SingleOrDefaultAsync(ac => ac.Id == htIdToMatch && ac.IsActive);
+
+            if (accommodation == default || accommodationToMatch == default)
+                return Result.Failure("Wrong accommodation Id");
+
+            foreach (var supplierAccommodation in accommodationToMatch.SupplierAccommodationCodes)
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs:133]: This is hard to read for me without braces

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs [hunk 3]
```diff
@@ -116,7 +118,80 @@ public async Task<Result> RecalculateData(int id)
...
+            };
+
+            var htAccommodationMappingToDeactivate = await
+                _context.HtAccommodationMappings
+                    .Where(hm => hm.HtId == htIdToMap && hm.IsActive)
+                    .SingleOrDefaultAsync();
+
+            if (htAccommodationMappingToDeactivate != default)
+            {
+                dbHtAccommodationMapping.MappedHtIds.UnionWith(htAccommodationMappingToDeactivate.MappedHtIds);
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs:172]: Looks like this space is not needed

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs [hunk 4]
```diff
@@ -30,38 +31,39 @@ public async Task<Result> MatchUncertain(int uncertainMatchId)
...
-        
+        public async Task<Result> MatchAccommodations(int htId, int htIdToMatch)
+        {
+            var (_, isFailure, error) = await Match(htId, htIdToMatch);
+            if (isFailure)
+                return Result.Failure(error);
+
+            await AddOrUpdateMappings(htId, htIdToMatch);
+            await _context.SaveChangesAsync();
+            // No need to check for failure
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs:61]: Add a reason why there is no need to the comment

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs]: Not understand, which space dou you mean?

**Review by kirillta — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs (hunk 3 above)
Comment (mingazhev) [reply HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs]: Saw a space there, now don't

**Review by mingazhev — APPROVED** [review]

**Pull request merged**